lambda_function.py

```diff
@@ -50,18 +48,18 @@ def verify_response(response: BaseHTTPResponse) -> str:
 
 
 def lambda_handler(event, context):
-    webhook_secret_name = os.environ["WEBHOOK_SECRET_NAME"]
+    webhook_parameter_name = os.environ["WEBHOOK_PARAMETER_NAME"]
     records: list[dict] = event["Records"]
 
     session = boto3.session.Session()
-    session_client: BaseClient = session.client(service_name="secretsmanager", region_name="eu-west-2")
+    session_client: BaseClient = session.client(service_name="ssm", region_name="eu-west-2")
 
     for record in records:
         slack_message = get_entity_info_and_return_slack_message(record)
         slack_message_in_json = {"text": slack_message}
 
         encoded_json_string: bytes = json.dumps(slack_message_in_json, indent=2).encode("utf-8")
-        slack_webhook_url = get_slack_webhook_url(session_client, webhook_secret_name)
+        slack_webhook_url = get_slack_webhook_url(session_client, webhook_parameter_name)
```

Comment:
Is this always the same webhook? If so it would be better to fetch this when the lambda first cold starts and hold it in a variable outside the handler function. At present we're potentially fetching this parameter multiple times in each invocation, depending on the batch configured.

Reply:
👍. That's a good idea.

Reply:
I think you've moved the wrong bit. We're fine to read the environment variable within each invocation if necessary but doing `get_slack_webhook_url` is the asynchronous API call. If you could do exactly what you've done for `webhook_secret_name` but instead for `slack_webhook_url` we'll only be making 1 API call to SSM per lambda container.

Reply:
🤦 Ah, I see what you mean; sorry, I read your comment quickly. I'll move the call out of the for loop and also save the webhook globally.